source/code/plugins/patch_management_lib.rb

```diff
@@ -12,28 +12,48 @@ class LinuxUpdates
     @@delimiter = "_"
     @@force_send_last_upload = Time.now
     MAJOR_MINOR_VERSION_REGEX = /([^\.]+)\.([^\.]+).*/
-
-    @@agentDetailsMap = Hash[(`cat /etc/opt/microsoft/omsagent/conf/omsadmin.conf`.split(/\r?\n/).map {|s| 
-                                s.split("=")}).map {|key, value| 
-                                    [key, value]
-                                }
-                            ]
-
-    @@hostOSDetailsMap = Hash[(`cat /etc/opt/microsoft/scx/conf/scx-release`.split(/\r?\n/).map {|s|
-                               s.split("=")}).map {|key, value|
-                                    [key, value]
-                                }
-                            ]
+    OMS_ADMIN_FILE = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+    SCX_RELEASE_FILE = "/etc/opt/microsoft/scx/conf/scx-release"
+
+    @@agentDetailsMap = if File.exist?(OMS_ADMIN_FILE)
+                            Hash[ File.open(OMS_ADMIN_FILE, "r") do |f|
+                                    f.each_line do |line|
+                                        line.split(/\r?\n/).map {|s| s.split("=")}.map {|key, value| [key, value]}
+                                    end
+                                end ]
+                        end
+
+    @@hostOSDetailsMap = if File.exist?(SCX_RELEASE_FILE)
+                            Hash[ File.open(SCX_RELEASE_FILE, "r") do |f|
+                                    f.each_line do |line|
+                                        line.split(/\r?\n/).map {|s| s.split("=")}.map {|key, value| [key, value]}
```

Comment:
You may want to do error handling : if the file is not readable, if the line does not contain "="